Ecoli 0.2.0

KennetLindquist opened the pull request:
### Main improvements in this PR:
Curations based on @cdebuck findings:
- Removed reaction XYLI2

Curations based on [Bernstein 2023](https://pmc.ncbi.nlm.nih.gov/articles/PMC10698504/):
- Made GHMT2r irreversible

Curations based on [Mao 2022](https://pmc.ncbi.nlm.nih.gov/articles/PMC8773657/):
- Made changes to GPR rules (summarized in issue https://github.com/SysBioChalmers/Ecoli-GEM/issues/25)

Curations based on [Carrao 2024](https://arxiv.org/abs/2406.16596):
- Made pntAB (THD2pp) stoichiometric coefficient of protons 1 instead of 2
- (Re-)Assign gene protein rule of glxK (b0514) to GLYCK2 and remove GLYCK
- Changed direction of HSDy (irreversible towards the homoserine production direction)
- Changed direction of SUCCt1pp (irreversible towards the export direction)

**I hereby confirm that I have:**

- [x] Tested my code on my own computer for running the model

KennetLindquist committed:
feat (model): curations Carrao 2024

feat (model): curations Carrao 2024, including python code for model changes

KennetLindquist committed:
chore: add macaw test result

KennetLindquist committed:
feat (model): remove GLYCK

feat (model): remove GLYCK and correct previous GLYCK2 gene reaction rule b3124 removal

KennetLindquist committed:
chore: add macaw test result

KennetLindquist committed:
fix (model): improves indentation consistency

fix (model): improves model indentation consistency (automatically through with cobrapy)

KennetLindquist committed:
feat (model): curations Mao 2022

feat (model): curations Mao 2022

KennetLindquist committed:
fix (model): improves indentation consistency

fix (model): improves model indentation consistency (automatically through cobrapy)

KennetLindquist committed:
feat (model): curation Bernstein 2023

feat (model): curation Bernstein 2023

KennetLindquist committed:
chore: add macaw test result

KennetLindquist committed:
fix (model): improves indentation consistency

fix (model): improves model indentation consistency (automatically through with cobrapy)

KennetLindquist committed:
feat (model): curation XYLI2 removal

feat (model): curation XYLI2 removal

KennetLindquist committed:
chore: add macaw test result

KennetLindquist committed:
feat: XYLI2 removal investigation

feat: XYLI2 removal investigation on impact on related metabolites. Conclusion: No dead-end reactions produced -> keep all metabolites involved in XYLI2

KennetLindquist committed:
feat (model): XYLI1 annotation update

feat (model): XYLI1 annotation update

KennetLindquist committed:
feat: GLYCK removal investigation

feat: GLYCK removal investigation on impact on related metabolites. Conclusion: No dead-end reactions produced -> keep all metabolites involved in GLYCK

KennetLindquist committed:
feat: complete collection of deprications

feat: complete collection of deprications

KennetLindquist committed:
feat (model): curation XYLI2 removal

### Main improvements in this PR:
- Implemented changes suggested by @cdebuck as outlined in issue #29  
- Update annotation information for XYLI1 found to be outdated in issue #31 

**I hereby confirm that I have:**

- [x] Tested my code on my own computer for running the model
- [x] Selected `develop` as a target branch

KennetLindquist committed:
feat (model): curation Bernstein 2023

### Main improvements in this PR:
- Implemented change from Bernstein 2023 as outlined in issue #27  

**I hereby confirm that I have:**

- [x] Tested my code on my own computer for running the model
- [x] Selected `develop` as a target branch

KennetLindquist committed:
feat (model): curations Mao 2022

### Main improvements in this PR:
- Implemented changes from Mao 2022 as outlined in issue #25 

**I hereby confirm that I have:**

- [x] Tested my code on my own computer for running the model
- [x] Selected `develop` as a target branch

KennetLindquist committed:
chore: merge latest changes from develop

KennetLindquist committed:
chore: move deprecated reactions to model/reactions folder

chore: move deprecated reactions to model/reactions folder

KennetLindquist committed:
chore: fix spelling

chore: fix spelling

KennetLindquist committed:
feat: add Carrao deprecated reaction to deprecated file

feat: add Carrao deprecated reaction to deprecated file

KennetLindquist committed:
feat (model): curations Carrao 2024

### Main improvements in this PR:
- Implemented changes from Carrao 2024 as outlined in issue #21 

**I hereby confirm that I have:**

- [x] Tested my code on my own computer for running the model
- [x] Selected `develop` as a target branch

mihai-sysbio commented on .deprecated/deprecated_complete_collection.yml:
```diff
@@ -0,0 +1,75 @@
+    id: "Ecoli-GEM"
+    name: "The Consensus Genome-Scale Metabolic Model of Escherichia coli"
+    date: "2025-06-16"
+    version: ""
```
How about adding a suggestive name for this file?
```suggestion
    version: "deprecated"
```

KennetLindquist committed:
chore: removed duplicate deprications

chore: removed duplicate deprications as they are stored in depricated_complete_collection

mihai-sysbio commented:
A note on the missing `.tsv` files reported by the GitHub Action - this is expected as these files are meant to be added in a future release. For now, the annotation is still part of the `yml` (and other file formats where applicable).
